src/ripple/nodestore/Types.h

```diff
@@ -53,6 +53,21 @@ enum Status {
 /** A batch of NodeObjects to write at once. */
 using Batch = std::vector<std::shared_ptr<NodeObject>>;
 
+/** Shard states. */
+enum class State {
+    acquire,     // Being acquired
+    complete,    // Backend contains all ledgers, requires finalizing
+    finalizing,  // Being finalized
+    final,       // Database verified, shard is immutable
+    queued       // Queued to be finalized
+};
+
+static constexpr State acquire = State::acquire;
```

Comment:
One motivation for doing this is that `enum class` does two things at once.  It both...

1. Makes an `enum` that doesn't turn into in `int` if you look at it wrong, and
2. Mandates that members of the `enum` be scoped.

The `enum` is already trapped in the `NodeStore` namespace.  So explicitly having the `State` enumerations leak out into the `NodeStore` namespace may give them adequate scoping.  The `enum class` still provides the benefit that a `State::acquire` won't casually devolve into an `int`.

I've used this idiom myself in places where I want enumeration values to be easy to access and still not be interchangeable with integers.

Reply:
That said, I ran the experiment of removing the `static constexpr` declarations to see how bad that damage is.  Personally, I found the code more readable using a text or web-based interface if it said `State::acquire` instead of just `acquire`.  There was one place in PeerImp.cpp where things got a bit verbose.  It looked like:
```
                case NodeStore::State::acquire:
                case NodeStore::State::complete:
                case NodeStore::State::finalizing:
                case NodeStore::State::queued:
```
Given that experiment I have the following suggestions:

- Rename the `enum` from `State` to `ShardState`,
- Move it out from the `NodeStore` namespace and into the enclosing `ripple` namespace, and
- Omit the `static constexpr` declarations.

I think that may result in the best readability, at least for web- and text-based viewing.